### ISSUE TITLE
from_dict causes "is incompatible with" type error 

### ISSUE THREAD

**Issue by davetapley** [yukinarit/pyserde#324]
In [pyright](https://github.com/microsoft/pyright):

```py
@serde
@dataclass
class Sensor:
    id: str

# This is okay ✅
sensor: Sensor = from_json(Sensor, '{"id": "123"}')

# This is not ❌
sensor: Sensor = from_dict(Sensor, {'id': '123'})
```

Latter causes:
```
Expression of type "list[Unknown | list[Any] | list[Unknown] | set[Any] | frozenset[Unknown] | set[Unknown] | tuple[Any, ...] | tuple[Unknown, ...] | dict[Any, Any] | defaultdict[Unknown, Any | Unknown] | dict[Unknown, Unknown] | NDArray[Unknown] | Any | None]" cannot be assigned to declared type "list[Sensor]"
  Type "Unknown | list[Any] | list[Unknown] | set[Any] | frozenset[Unknown] | set[Unknown] | tuple[Any, ...] | tuple[Unknown, ...] | dict[Any, Any] | defaultdict[Unknown, Any | Unknown] | dict[Unknown, Unknown] | NDArray[Unknown] | Any | None" cannot be assigned to type "Sensor"
    "dict[Any, Any]" is incompatible with "Sensor"
```

--- 

I discovered this via workaround for:
- https://github.com/yukinarit/pyserde/discussions/323

**Comment by yukinarit** [comment]
Hi @davetapley 
Thanks, let me take a look

**Comment by yukinarit** [comment]
@davetapley Fixed in v0.10.2 👍
